### PR TITLE
Demystify the convert method

### PR DESCRIPTION
**Pull request by suricactus**
There were a series of mistakes lately, caused by the overcluttering the `convert` method. Even the linter was saying it was overly complex. At least huge block of code are now named, but there is the potential for even more.

- moved the basemap requirements check to standalone function
- don't break on failed basemap generation
- use the convention "o_" for "original"/online and "e_" for "exported"/offline
- move layer postprocessing to standalone function
- drop support for QGIS <3.16.1

### CODE REVIEW

**Comment by m-kuhn** [comment]
> drop support for QGIS 3.16

or "drop support for QGIS < 3.16.1"?

**Comment by m-kuhn** [comment]
Can you make sure the plugin itself will reflect this requirement in the metadata.txt when updated?

File: offline_converter.py [hunk 1]
```diff
@@ -94,467 +111,406 @@ def __init__(
...
-            copied_files = list()
-            for current_layer_index, layer in enumerate(self.__layers):
-                self.total_progress_updated.emit(
-                    current_layer_index - len(self.__offline_layers),
-                    len(self.__layers),
-                    self.trUtf8("Copying layers…"),
+                self.post_process_fields(e_layer)
+
+    def post_process_fields(self, e_layer: QgsVectorLayer) -> None:
+        QgsProject.instance()
```
Comment (m-kuhn) [inline offline_converter.py:393]: What's that for?

**Review by m-kuhn — APPROVED** [review]
very valuable refactoring, thanks

File: offline_converter.py (hunk 1 above)
Comment (suricactus) [reply offline_converter.py:393]: oops

**Comment by suricactus** [comment]
Thanks a lot, @m-kuhn for the review, much appreciated!

**Pull request merged**